File: Tools/skyConverter/skyConverter/ModelStructure/Mesh.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace skyConverter.ModelStructure
{
    class Mesh : ISerializer
    {
        private String _name = "";
	    private int _countVertices;
        private List<Face> _faces;
        private Material _material;

	    public Mesh(String _name)
        {
		    this._name = _name;
		    _faces = new List<Face>();

            this._material = new Material("Materials/Phong.mat", "PHONG");
            this._material.specular = new double[] { 1.0, 1.0, 1.0, 1.0 };
            this._material.specularIntensity = 0.0;

            this._material.diffuse = new double[] { 1.0, 1.0, 1.0, 1.0 };
	    }
	
	    public void AddFace(Face face)
        {
		    _faces.Add(face);
	    }

	    public void SetCountVertices()
        {
            this._countVertices = _faces.Count * 3;
	    }


	    public JObject serializeObject() 
        {
		    JObject obj = new JObject();
		
		    obj.Add("Name", _name);
		    obj.Add("Vertices", _countVertices);

            if (_faces.Count > 0)
            {
			    JArray facesNode = new JArray();

                for (int i = 0; i < _faces.Count; i++)
                {
                    facesNode.Add(_faces[i].serializeArray());
			    }

			    obj.Add("Faces", facesNode);
                obj.Add("Material", this._material.serializeObject());

			    return obj;
		    }



		    return null;
	    }

        public JArray serializeArray()
        {
		    return null;
	    }
    }
}

File: Tools/Converter/src/toJSON/Mesh.java
package toJSON;

import java.util.ArrayList;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;


public class Mesh implements ISerializer{

	private String name="";
	private int countVertices;
	private ArrayList<Face> faces;
	
	public Mesh(String _name){
		name = _name;
		
		faces = new ArrayList<Face>();
	}
	
	
	public void addFace(Face face){
		faces.add(face);
	}

	public void setCountVertices(int _countVertices){
		countVertices = _countVertices;
	}
	@Override
	public JSONObject serializeObject() {
		
		JSONObject obj = new JSONObject();
		
		obj.put("Name", name);
		obj.put("Vertices", countVertices);
		
		if(!faces.isEmpty()){
			JSONArray facesNode = new JSONArray();
			for(int i = 0; i < faces.size(); i++){
				facesNode.add(faces.get(i).serializeArray());
			}
			obj.put("Faces", facesNode);
			return obj;
		}
		return null;
	}


	@Override
	public JSONArray serializeArray() {
		return null;
	}
	
	
}
